### PR TITLE
Fix New Post Processing

### PR DESCRIPTION
**Pull request by fppt**
# Why is this PR needed?

The new PP works but only executes one job per keyspace per second. This means that graph with 10000 attributes will take 10000 seconds to completely post process. That is just dumb. 

# What does the PR do?

Submit multiple jobs for post processing per second rather than just one job.

# Does it break backwards compatibility?

No. 

# List of future improvements not on this PR

Smart job submission. Right now it submits however many jobs there are threads. Ideally it should submit jobs based on load.

### CODE REVIEW

File: grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java [hunk 1]
```diff
@@ -94,4 +94,34 @@ public void whenTheIndexCacheIsEmpty_NothingStarts(){
...
+        when(indexPostProcessor.popIds(keyspaceA, index1)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index2)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index3)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index4)).thenReturn(ids);
+
+        //Run the method
+        postProcessingTask.run();
+
+        //Give time for PP to run
+        Thread.sleep(config.getProperty(GraknConfigKey.POST_PROCESSOR_DELAY) * 2000);
```
Comment (aelred) [inline grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java]: Why 2000?

File: grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/PostProcessingTask.java [hunk 2]
```diff
@@ -47,23 +47,29 @@
...
-                threadPool.schedule(() -> processIndex(keyspace, index), postprocessingDelay, TimeUnit.SECONDS);
-            }
+            String index;
+            int limit = 0;
+            do{
+                 index = indexPostProcessor.popIndex(keyspace);
+                 final String i = index;
+                 if(index != null) threadPool.schedule(() -> processIndex(keyspace, i), postprocessingDelay, TimeUnit.SECONDS);
+                 limit++;
+            } while(index != null && limit < postProcessingMaxJobs);
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/PostProcessingTask.java]: Look at beautiful Java 9 possibilities:
```
Stream.generate(() -> indexPostProcessor.popIndex(keyspace))
   .takeWhile(index -> index != null)
   .limit(postProcessingMaxJobs)
   .forEach(index -> threadPool.schedule(() -> processIndex(keyspace, i), ppDelay, SECONDS));
```

File: grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java (hunk 1 above)
Comment (lolski) [reply grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java]: i think this is the part where we wait for cassandra index to be ready?

Comment (aelred) [reply grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java]: I'm just wondering about the exact number - sleeping in tests is a bit dangerous, so we should make sure we're sleeping for the right length.

File: grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/PostProcessingTask.java (hunk 2 above)
Comment (fppt) [reply grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/PostProcessingTask.java]: Oooooooooooo . . .  pretty

File: grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java (hunk 1 above)
Comment (fppt) [reply grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java]: Ya 2 seconds is just to give time for the index to stabilise. I agree that sleeping is dangerous but we have no other way to check the status of the index right now.

**Pull request merged**